macros/REST_Axion_PlotNgamma.C

```diff
@@ -0,0 +1,157 @@
+#include "TRestAxionField.h"
+#include "TRestAxionSolarFlux.h"
+#include "TRestAxionSolarModel.h"
+#include "TRestAxionSolarQCDFlux.h"
+//*******************************************************************************************************
+//*** Description: It computes the number of photons for each gas step for each axion mass
+
+//***
+//*** Arguments by default are (in order):
+//*** - *ma_max* = 0.1: Maximum axion mass (in eV) to be plotted.
+//*** - *ma_min* = 0: Minimum axion mass (in eV) to be plotted.
+//*** - *Ea* = 4.2: Axion energy (in keV).
+//*** - *Bmag* = 2.5: Magnetic field (in T).
+//*** - *Lcoh* = 10000: Coherence length (in mm).
+//*** - *gasName* = "He": Gas name.
+//*** - *vacuum* = true: If true, the vacuum probability is added to the sum of all the probabilities.
+//*** - *n_ma* = 10000: Number of points to be plotted.
+//*** - *E_a_min* = 0.0: Minimum energy (in keV) to be plotted.
+//*** - *E_a_max* = 20.0: Maximum energy (in keV) to be plotted.
+//***
+//*** The generated plots are the results from `TRestAxionField::GetMassDensityScanning`,
+//*** `TRestAxionField::GammaTransmissionFWHM`, `TRestAxionBufferGas::GetMassDensity`,
+//*** `TRestAxionField::GammaTransmissionProbability` and `TRestAxionSolarQCDFlux`.
+
+//***
+//*** --------------
+//*** Usage: restManager PlotNgamma [ma_max=0.1] [ma_min=0] [Ea=4.2] [Bmag=2.5] [Lcoh=10000] [gasName=He]
+
+//*** [vacuum=true] [n_ma=10000] [E_a_min=0.0] [E_a_max=20.0]
+//***
+//*** Being all of them optional arguments.
+//*** --------------
+//*** Author: Fran Candón
+//*******************************************************************************************************
+
+int REST_Axion_PlotNgamma(double ma_max = 0.1, double ma_min = 0, double Ea = 4.2, double Bmag = 2.5,
+
+                          double Lcoh = 10000, std::string gasName = "He", Bool_t vacuum = true,
+                          int n_ma = 100, double E_a_min = 0.0, double E_a_max = 20.0) {
+    // Creates the vector of axion masses
+    std::vector<double> m_a;
+    std::vector<double> n_photons;
+    std::vector<double> n_photons_gas;
+
+    double ma_step = (ma_max - ma_min) / n_ma;
+    for (int i = 0; i < n_ma; i++) {
+        m_a.push_back(ma_min + i * ma_step);
+        n_photons.push_back(0);
+        n_photons_gas.push_back(0);
+    }
+
+    // Creates the solar axion flux
+    TRestAxionSolarQCDFlux* sFlux = new TRestAxionSolarQCDFlux("fluxes.rml", "Gianotti");
+    sFlux->Initialize();
+    TH1F* spec1 = sFlux->GetEnergySpectrum();
+    spec1->Rebin(20000 / n_ma);

```

Comment:
You could simply multiply `spec1` bins by the results of the axion-photon probability at each energy.


for( int n = 0; n < spec1.GetNbinsX(); n++ )
{
   double en = spec1->GetBinCenter(n+1);
   // Get probability for `en` --> Paxion

   spec1->SetBinContent(n+1, spec1->GetBinContent(n+1) * Paxion );
}

// Then calculate the integral of `spec1`

Reply:
That’s great, I’ll change that as soon as I can. I had to rebin the spectrum because I was using this method of storing the Probabilty values in a histogram, and then multiply, then, both histograms needs to be the same size.